version: '3.9'

services:
    
    postgresdb:
        image: postgres:latest
        ports:
            - 5555:5432
        restart: on-failure
        volumes:
            - ./scripts/db:/docker-entrypoint-initdb.d        
            - ./postgres-data:/var/lib/postgresql/data
        environment:
            POSTGRES_PASSWORD: postgres
            POSTGRES_USER: postgres
            PGDATA: "/var/lib/postgresql/data/pgdata"
            
    cache:
        image: redis:7.0-alpine
        restart: on-failure
        ports:
            - '6379:6379'
        environment:
            - ALLOW_EMPTY_PASSWORD=yes
        volumes: 
            - ./redis-data:/data

    auth-service:
        build:
            context: ./
            dockerfile: ./server/AuthService/Dockerfile
        ports:
            - 9004:80
        restart: on-failure
        depends_on:
            - postgresdb

    process-quest-service:
        build:
            context: ./
            dockerfile: ./server/ProcessQuestService/Dockerfile
        ports:
            - 9007:80
        restart: on-failure
        depends_on:
            - cache   

    generate-quest-service:
        build:
            context: ./
            dockerfile: ./server/GenerateQuestsService/Dockerfile
        ports:
            - 9005:80
        restart: on-failure
        depends_on:
            - postgresdb        
    
    quest-core:
        build:
            context: ./
            dockerfile: ./server/QuestCore/Dockerfile
        ports:
            - 9003:80
        restart: on-failure
        depends_on:
            - postgresdb

volumes:
    postgresdb:
    cache: